arbos/l2pricing/l2pricing_constraints.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package l2pricing
+
+import (
+	"github.com/ethereum/go-ethereum/arbitrum/multigas"
+
+	"github.com/offchainlabs/nitro/arbos/constraints"
+	"github.com/offchainlabs/nitro/arbos/storage"
+)
+
+type L2PricingStateConstraints struct {
+	L2PricingState
+	constraintsCount storage.StorageBackedUint64
+	resourceBacklogs [multigas.NumResourceKind]storage.StorageBackedUint64
```

Comment:
The backlog will be deducted based on the constraint time period and target, so it is unique to the constraint. We can't associate it with a single resource, so, we probably should store the backlog inside the ResourceConstraints struct.

Reply:
That's a good point. I can move `resourceBacklogs` to `ResourceConstraints` and store them with `StorageBytes `